client/src/components/QualityManagement/Parts.jsx

```diff
@@ -0,0 +1,325 @@
+        </Heading>
+        <Center mt={4}>
+          <Button onClick={onOpen}>Add Defect</Button>
+        </Center>
+        {addDefectForm()}
+        <NoResultImage />
+      </>
+    );
+  }
+
+  return (
+    <Box overflowX="auto">
+      <IconButton
+        colorScheme="blue"
+        variant="outline"
+        aria-label="add"
+        float="right"
+        m={2}
+        icon={<SmallAddIcon />}
+        onClick={onOpen}
+      />
+      <IconButton
+        colorScheme="blue"
+        variant="outline"
+        aria-label="delete"
+        float="right"
+        m={2}
+        icon={<DeleteIcon />}
+        onClick={handleDelete}
+        isDisabled={selected.length === 0}
+      />
+      <TableContainer component={Paper}>
+        <Table minWidth="unset" width="100%" variant="striped" colorScheme="light">
+          <Thead>
+            <Tr>
+              <StyledTableHeader>
+                <Checkbox
+                  isIndeterminate={selected.length > 0 && selected.length < data.data.length}
+                  isChecked={data.data.length > 0 && selected.length === data.data.length}
+                  onChange={handleSelectAllClick}
+                >
+                  Ticket ID
+                </Checkbox>
+              </StyledTableHeader>
+              <StyledTableHeader>Product Name</StyledTableHeader>
+              <StyledTableHeader>Defect Type</StyledTableHeader>
+              <StyledTableHeader>Status of Defect</StyledTableHeader>
+              <StyledTableHeader>Desc of Solution</StyledTableHeader>
+            </Tr>
+          </Thead>
+          <Tbody>
+            {isSuccess &&
+              data.data
+                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                .map((defect) => (
+                  <Fragment key={defect._id}>
+                    <StyledTableRow>
+                      <StyledTableCell>
+                        <Checkbox
+                          onChange={(event) => handleOnChange(event, defect._id)}
+                          isChecked={isSelected(defect._id)}
+                        >
+                          {defect.id}
+                        </Checkbox>
+                      </StyledTableCell>
+                      <StyledTableCell>{defect.partName}</StyledTableCell>
+                      <StyledTableCell>
+                        <Tag
+                          size="md"
+                          variant="solid"
+                          colorScheme={defect.type === 'Broken' ? 'red' : 'orange'}
+                        >
+                          {defect.type}
+                        </Tag>
+                      </StyledTableCell>
+                      <StyledTableCell>
+                        <Tag
+                          size="md"
+                          variant="solid"
+                          colorScheme={handleStatusColor(defect.status)}
+                        >
+                          {defect.status}
```

Comment:
should we allow the user to change the detect status?

Reply:
I am not sure what do you mean by 'roles'

Reply:
I am referring to this column

![image](https://user-images.githubusercontent.com/60051944/111508940-9b5af780-8722-11eb-8d95-c1a994b5cff4.png)